src/app/games/awale/AwaleRules.ts

```diff
@@ -2,29 +2,33 @@ import { GameStatus, Rules } from '../../jscaip/Rules';
 import { MGPNode } from 'src/app/jscaip/MGPNode';
 import { AwaleState } from './AwaleState';
 import { AwaleMove } from './AwaleMove';
-import { AwaleLegalityStatus } from './AwaleLegalityStatus';
-import { ArrayUtils } from 'src/app/utils/ArrayUtils';
+import { ArrayUtils, Table } from 'src/app/utils/ArrayUtils';
 import { display } from 'src/app/utils/utils';
-import { MGPValidation } from 'src/app/utils/MGPValidation';
 import { Coord } from 'src/app/jscaip/Coord';
 import { AwaleFailure } from './AwaleFailure';
+import { MGPFallible } from 'src/app/utils/MGPFallible';
 
-export class AwaleNode extends MGPNode<AwaleRules, AwaleMove, AwaleState, AwaleLegalityStatus> {}
+export type AwaleLegalityInformation = {
```

Comment:
AwaleMoveResult ?

Reply:
ça me semble mieux effectivement, je généralise à tous les *LegalityInformation ?

Reply:
La ou sur awale le isLegal donne le move result, sur tout les jeux c'est les information qu'on obtient en calculant la legalité du mouvement, donc LegalityInformation "a un joli anneau à lui"... Gardons l'état actuel du coup ?

Reply:
oOk! Mais du coup j'ai changé pour l'awalé

Reply:
a nice ring to it :D !

Reply:
But no, let's uniformise prego :D !